src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,56 @@
+package core.basesyntax;
+
+import java.util.Locale;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private final Class[] figure = {Circle.class, Square.class, RightTriangle.class,
+            Rectangle.class, IsoscelesTrapezoid.class};
+
+    public Figure getRandomFigure() {
+        int indexPossibleClass = random.nextInt(figure.length);
+        Class randomClass = figure[indexPossibleClass];
+        switch (randomClass.getSimpleName()) {
+            case "Circle":
+                return generateNewCircle();

```

Comment:
' Class randomClass = figure[indexPossibleClass];' is redundant, remove it.
refactor using the next approach:

```suggestion
        int figureIndex = random.nextInt(figure.length);
        switch (figureIndex) {
            case 0:
                return generateNewCircle();
...
```